lib/Transforms/Scalar/LoadCombine.cpp

```diff
@@ -186,7 +186,7 @@ bool LoadCombine::combineLoads(SmallVectorImpl<LoadPOPPair> &Loads) {
 
   // Find first load. This is where we put the new load.
   LoadPOPPair FirstLP;
-  FirstLP.InsertOrder = -1u;
+  FirstLP.InsertOrder = std::numeric_limits<unsigned>::max();
```

Comment:
Why convert -1u to `std::numeric_limits<unsigned>::max();`, when above you converted -1U to `~0U`?